src/app/ReadClient.h

```diff
@@ -288,6 +304,8 @@ class ReadClient : public Messaging::ExchangeDelegate
 
     void OnUnsolicitedReportData(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle && aPayload);
 
+    void OnUnsolicitedCommunication() { mpCallback.OnUnsolicitedCommunication(this); }
```

Comment:
Should this method take the session the communication came in on as an argument?  And if this read client is currently in a "I am waiting to resubscribe" state, should it shift over to that new session?

And should that session be propagated on to the consumers, so they can do something useful with it (like start a new subscription) if they want to?

Reply:
Given the API is not calling the application on the Matter queue, the session won't be usable anyway, and so whatever the application want to do based on these callbacks will go through the MTRDeviceController's `getSessionForNode:` path and get the session.

Reply:
I meant in terms of MTRDevice itself (or equivalent for other consumers) or the ReadClient to use the session.  But I think it will be OK if people just use the CASE session manager: it should give them that exact session, synchronously, since it's the one with the most recent activity for the peer.